Sources/NIO/PipePair.swift

```diff
@@ -0,0 +1,137 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2019 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+extension NIOFileHandle: Selectable {
+}
+
+final class PipePair: SocketProtocol {
+    typealias SelectableType = NIOFileHandle
+
+    let inputFD: NIOFileHandle
+    let outputFD: NIOFileHandle
+
+    init(inputFD: NIOFileHandle, outputFD: NIOFileHandle) throws {
+        self.inputFD = inputFD
+        self.outputFD = outputFD
+        for fileHandle in [inputFD, outputFD] {
+            try fileHandle.withUnsafeFileDescriptor {
+                try NIOFileHandle.setNonBlocking(fileDescriptor: $0)
+                try ignoreSIGPIPE(descriptor: $0)
+            }
+        }
+    }
+
+    var description: String {
+        return "PipePair { in=\(inputFD), out=\(outputFD) }"
+    }
+
+    func connect(to address: SocketAddress) throws -> Bool {
+        return true
```

Comment:
Why isn't this a `ChannelError.operationUnsupported`?

Reply:
thanks, no reason I think, fixing.